core/src/fields.rs

```diff
@@ -469,50 +469,150 @@ impl MongoFields {
+                    .unwrap(),
+            );
         }
     }
 }
 
 impl MongoStatement for MongoFields {
     // Move the cursor to the next document and update the current row.
     // Return true if moving was successful, false otherwise.
-    fn next(&mut self, _mongo_connection: Option<&MongoConnection>) -> Result<bool> {
-        unimplemented!()
+    fn next(&mut self, mongo_connection: Option<&MongoConnection>) -> Result<bool> {
+        self.current_field_for_collection += 1;
+        Ok(
+            (self.current_field_for_collection as usize) < self.current_col_metadata.len()
+                || self.get_next_metadata(mongo_connection.unwrap()),
+        )
     }
 
     // Get the BSON value for the cell at the given colIndex on the current row.
     // Fails if the first row as not been retrieved (next must be called at least once before getValue).
-    fn get_value(&self, _col_index: u16) -> Result<Option<Bson>> {
-        unimplemented!()
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
+        // use self.current_col_metadata[current_field_for_collection]
+        // 1 -> TABLE_CAT
+        // 2 -> TABLE_SCHEM  (NULL)
+        // 3 -> TABLE_NAME
+        // 4 -> COLUMN_NAME
+        // 5 -> DATA_TYPE
+        // 6 -> TYPE_NAME
+        // 7 -> COLUMN_SIZE
+        // 8 -> BUFFER_LENGTH
+        // 9 -> DECIMAL_DIGITS
+        // 10 -> NUM_PREC_RADIX
+        // 11 -> NULLABLE
+        // 12 -> REMARKS
+        // 13 -> COLUMN_DEF
+        // 14 -> SQL_DATA_TYPE
+        // 15 -> SQL_DATETIME_SUB
+        // 16 -> CHAR_OCTET_LENGTH
+        // 17 -> ORDINAL_POSITION
+        // 18 -> IS_NULLABLE "YES" or "NO"
+        let get_meta_data = || {
+            self.current_col_metadata
+                .get(self.current_field_for_collection)
+                .unwrap()
+        };
+        Ok(Some(match col_index {
+            1 => Bson::String(self.current_db_name.clone()),
+            2 => Bson::Null,
+            3 => Bson::String(get_meta_data().table_name.clone()),
+            4 => Bson::String(get_meta_data().col_name.clone()),
+            5 => Bson::Int32(get_meta_data().sql_type.0 as i32),
+            6 => Bson::String(get_meta_data().type_name.clone()),
+            7 => Bson::Int32(get_meta_data().length.unwrap_or(0) as i32),
+            8 => Bson::Int32(get_meta_data().length.unwrap_or(0) as i32),
+            9 => Bson::Int32(get_meta_data().precision.unwrap_or(0) as i32),
+            10 => match get_meta_data().sql_type {
+                SqlDataType::INTEGER | SqlDataType::DOUBLE => Bson::Int32(2),
+                SqlDataType::DECIMAL => Bson::Int32(10),
+                _ => Bson::Null,
+            },
+            11 => Bson::Int32(get_meta_data().nullability.0 as i32),
+            12 => Bson::String("".to_string()),
+            13 => Bson::Null,
+            14 => Bson::Int32(get_meta_data().sql_type.0 as i32),
+            15 => Bson::Null,
```

Comment:
If the data type is TIMESTAMP, then this is SQL_CODE_TIMESTAMP (3) otherwise it is null.
We should probably add this in BsonTypeInfo too, so that we won't have to duplicate the logic when/if we implement SQLGetDescRec.